docs/quickstart/expansion.rst

```diff
@@ -0,0 +1,185 @@
+Cluster expansion
+=================
+
+.. _`kubeadm join`:
+      https://kubernetes.io/docs/reference/setup-tools/kubeadm/kubeadm-join/
+.. _`bootstrap tokens`:
+      https://kubernetes.io/docs/reference/command-line-tools-reference/kubelet-tls-bootstrapping/
+
+.. |kubeadm join| replace:: ``kubeadm join``
+
+Now that the :term:`Bootstrap node` has been installed
+(see :doc:`./bootstrap`), the cluster can grow from there.
+Unlike the |kubeadm join|_ approach which relies on `bootstrap tokens`_ and
+manual operations on each node, MetalK8s uses Salt SSH to setup new
+:term:`Nodes <Node>` through declarative configuration,
+from a single entrypoint. You can choose to either
+:ref:`use the UI <quickstart-expansion-ui>` or
+:ref:`the command-line <quickstart-expansion-cli>`
+to perform the cluster expansion.
+
+Defining an architecture
+------------------------
+
+.. todo::
+
+   remind roles from intro, chosen arch: 3 control-plane + etcd, 2 workers
+   (one being dedicated for infra)
+
+
+.. _quickstart-expansion-ui:
+
+Adding a node with the :ref:`quickstart-services-admin-ui`
+----------------------------------------------------------
+
+.. todo::
+
+   - node declaration
+   - deployment
+   - troubleshooting (example errors)
+
+
+.. _quickstart-expansion-cli:
+
+Adding a node from the command-line
+-----------------------------------
+
+Creating a manifest
+^^^^^^^^^^^^^^^^^^^
+For each node to add, you will need to create a
+:term:`manifest <Node manifest>` file, following the template below:
+
+.. code-block:: yaml
+
+   apiVersion: v1
+   kind: Node
+   metadata:
+     name: <node name>
+     annotations:
+       metalk8s.scality.com/ssh-key-path: /etc/metalk8s/pki/bootstrap
+       metalk8s.scality.com/ssh-host: <node control-plane IP>
+       metalk8s.scality.com/ssh-sudo: 'false'
+     labels:
+       metalk8s.scality.com/version: '2.0'
+       <role labels>
+   spec:
+     taints: <taints>
+
+The combination of ``<role labels>`` and ``<taints>`` will determine what is
+installed and deployed on the Node.
+
+A node exclusively in the control-plane with ``etcd`` storage will have:
+
+.. code-block:: yaml
+
```

Comment:
Let's add ellipses (`[…]`) to indicate this is part of a node manifest above the snippet, between `metadata` and `spec`, and below the snippet.